presto-hive/src/main/java/io/prestosql/plugin/hive/s3/PrestoS3FileSystem.java

```diff
@@ -939,35 +1009,54 @@ private InputStream openStream(Path path, long start)
                             }
                         });
             }
-            catch (InterruptedException e) {
-                Thread.currentThread().interrupt();
-                throw new RuntimeException(e);
-            }
             catch (Exception e) {
-                throwIfInstanceOf(e, IOException.class);
-                throwIfUnchecked(e);
-                throw new RuntimeException(e);
+                throw propagate(e);
             }
         }
 
         private void closeStream()
         {
             if (in != null) {
-                try {
-                    if (in instanceof S3ObjectInputStream) {
-                        ((S3ObjectInputStream) in).abort();
-                    }
-                    else {
-                        in.close();
-                    }
-                }
-                catch (IOException | AbortedException ignored) {
-                    // thrown if the current thread is in the interrupted state
-                }
+                abortStream(in);
                 in = null;
                 STATS.connectionReleased();
             }
         }
+
+        private void checkClosed()
+                throws IOException
+        {
+            if (closed.get()) {
+                throw new EOFException(FSExceptionMessages.NEGATIVE_SEEK);
+            }
+        }
+
+        private static void abortStream(InputStream in)
+        {
+            try {
+                if (in instanceof S3ObjectInputStream) {
+                    ((S3ObjectInputStream) in).abort();
+                }
+                else {
+                    in.close();
+                }
+            }
+            catch (IOException | AbortedException ignored) {
+                // thrown if the current thread is in the interrupted state
```

Comment:
Log exception at least? Don't we need to set `interrupted` state?

Reply:
I don't see any point in logging it. You will just get a bunch of useless exceptions in the log. This happens a lot when interrupt driver threads (say for a `LIMIT` query or when a query is cancelled).

Reply:
The AWS library should not be resetting the interrupted state, but it appears that it does in various places. That's a bug and should be fixed in the library. Working around it on our end would be much more involved.